CMakeLists.txt

```diff
@@ -11,24 +11,29 @@ cmake_minimum_required (VERSION 3.14)
 
 # check if this is the main project
 if (NOT DEFINED PROJECT_NAME)
+    set(ROOT_PROJECT TRUE)
+endif ()
+
+project(ivio LANGUAGES CXX VERSION "${IVIO_PROJECT_VERSION}"
+            DESCRIPTION "ivio -- IO library for file formats used in bioinformatics")
+
+if (${ROOT_PROJECT})
     set(CMAKE_CXX_STANDARD 20)
     set(CMAKE_CXX_STANDARD_REQUIRED ON)
     set(CMAKE_CXX_EXTENSIONS OFF)
+
     if(WIN32)
         set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /bigobj /EHsc -DNOMINMAX")
-    else()
+    elseif (NOT CMAKE_CXX_COMPILER_ID STREQUAL "IntelLLVM")
```

Comment:
Just so you know, cmake only detects IntelLLVM starting with some version (don't know which), since this is a relatively new compiler.

Before that, it is detected as Clang.

Reply:
Should detect it with cmake >= 3.20

Reply:
So, to be technically correct, one would need to either bump `cmake_minimum_required` to 3.20, or do a CXX check in CMake for whether `__INTEL_LLVM_COMPILER` is defined if CMake < 3.20.

Both are kind of meh.

I would just add a sentence to the README that says `Intel oneAPI C++ Compiler (IntelLLVM) requires CMake >= 3.20`.

Reply:
Thank you,  I will bump it up to 3.20